examples/seg/deeplabv3/deeplabv3.py

```diff
@@ -0,0 +1,257 @@
+        self.backbone = backbone
+        self.aspp = ASPP(
+            atrous_rates=[1, 6, 12, 18],
+            is_training=is_training,
+            in_channels=2048,
+            num_classes=args.num_classes,
+        )
+        self.conv = nn.Conv2d(256, args.num_classes, kernel_size=1, weight_init="xavier_uniform", has_bias=True)
+
+    def construct(self, x):
+        size = ops.shape(x)
+        features = self.backbone(x)[-1]
+        out = self.aspp(features)
+        out = self.conv(out)
+        out = ops.interpolate(out, size=(size[2], size[3]), mode="bilinear", align_corners=True)
+        return out
+
+
+class Decoder(nn.Cell):
+    """
+    Decoder module of DeepLabV3P model.
+
+    """
+
+    def __init__(
+        self,
+        low_level_channels: int = 256,
+        num_classes: int = 21,
+        weight_init: str = "xavier_uniform",
+    ):
+        super(Decoder, self).__init__()
+        self.conv1 = nn.Conv2d(low_level_channels, 48, kernel_size=1, weight_init=weight_init)
+        self.bn1 = nn.BatchNorm2d(48)
+        self.relu = nn.ReLU()
+
+        self.last_conv = nn.SequentialCell(
+            [
+                conv3x3(304, 256, stride=1, dilation=1, padding=1),
+                nn.BatchNorm2d(256),
+                nn.ReLU(),
+                conv3x3(256, 256, stride=1, dilation=1, padding=1),
+                nn.BatchNorm2d(256),
+                nn.ReLU(),
+                conv1x1(256, num_classes, stride=1),
+            ]
+        )
+
+    def construct(self, x, low_level_features):
+        size = ops.shape(low_level_features)
+        low_level_features = self.conv1(low_level_features)
+        low_level_features = self.bn1(low_level_features)
+        low_level_features = self.relu(low_level_features)
+
+        x = ops.ResizeNearestNeighbor(size=(size[2], size[3]), align_corners=True)(x)
+        x = ops.cat((x, low_level_features), axis=1)
+        x = self.last_conv(x)
+
+        return x
+
+
+class DeeplabV3Plus(nn.Cell):
+    """
+    DeepLabV3Plus implement.
+    """
+
+    def __init__(
+        self,
+        backbone,
+        args,
+        is_training: bool = True,
+    ):
+        super(DeeplabV3Plus, self).__init__()
+        self.is_training = is_training
+        self.backbone = backbone
+        self.aspp = ASPP(
+            atrous_rates=[1, 6, 12, 18],
+            is_training=is_training,
+            in_channels=2048,
+            num_classes=args.num_classes,
+        )
+        # low_level_channels : Resnet-256, Xception-128
+        self.decoder = Decoder(low_level_channels=256, num_classes=args.num_classes)
+
+    def construct(self, x):
+        size = ops.shape(x)
+        low_level_features, features = self.backbone(x)
+        out = self.aspp(features)
+        out = self.decoder(out, low_level_features)
+        out = ops.interpolate(out, size=(size[2], size[3]), mode="bilinear", align_corners=True)
+        return out
+
+
+class BuildInferNetwork(nn.Cell):
```

Comment:
Weird name. and why we need it

Reply:
rename to DeepLabInferNetwork. 
DeepLabV3 and DeepLabV3Plus share the same infer network.